src/web/middleware/precondition.rs

```diff
@@ -158,7 +139,18 @@ where
             let bso = BsoParam::extrude(sreq.head(), &mut sreq.extensions_mut()).ok();
             let bso_opt = bso.map(|b| b.bso);
 
-            let resource_ts = db.extract_resource(user_id, collection, bso_opt).await?;
+            let (req, payload) = sreq.into_parts();
```

Comment:
Very nice way to work around the extractor not usually having access to the ServiceRequest, and then still turning it back into sreq later. Nice.

Opened a new bug to move this into a plain function at some point. https://github.com/mozilla-services/syncstorage-rs/issues/711

Reply:
There's actually an issue open to avoid the possible panic when we recombine the parts: #695